common/util.go

```diff
@@ -410,3 +411,47 @@ func ReadAtMost(r io.Reader, n int) ([]byte, error) {
 	}
 	return b, err
 }
+
+func enumerateNvidiaDevices() []string {
+	gpu, err := ghw.GPU()
+	if err != nil {
+		glog.Errorf("Error initializing GPU detection via ghw: %v", err)
+		return nil
+	}
+
+	nvidiaCardCount := 0
+	for _, card := range gpu.GraphicsCards {
+		if strings.EqualFold(card.DeviceInfo.Vendor.Name[:6], "nvidia") {
+			nvidiaCardCount += 1
+		}
+	}
+	if nvidiaCardCount == 0 {
+		glog.Warning("No Nvidia devices detected")
+		return nil
+	}
+
+	devices := []string{}
+
+	for i := 0; i < nvidiaCardCount; i++ {
+		s := strconv.Itoa(i)
+		devices = append(devices, s)
+	}
+
+	return devices
+}
+
+func ParseNvidiaDevicesFlag(nvidia string) []string {
```

Comment:
I suggest dropping the "Flag" as we can treat this as a function that parses a string of Nvidia devices that *could* come from a flag, but does not necessarily need to be from a flag.